assets/src/edit-story/components/form/dateTime/timePicker.js

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled, { css } from 'styled-components';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+const TimeWrapper = styled.div`
+  margin-bottom: 1em;
+  padding: 0 20px;
+`;
+
+const Fieldset = styled.fieldset`
+  margin-top: 0.5em;
+  position: relative;
+`;
+
+const Legend = styled.legend`
+  position: absolute;
+  top: -999em;
+  left: -999em;
+`;
+
+const InputRow = styled.div`
+  display: flex;
+`;
+
+const inputCSS = css`
+  min-height: 30px;
+  box-shadow: 0 0 0 transparent;
+  transition: box-shadow 0.1s linear;
+  border-radius: 4px;
+  border: 1px solid #7e8993;
+  box-sizing: border-box;
+  height: 28px;
+  vertical-align: middle;
+  background-color: #fff;
+  color: #32373c;
+`;
+
+const NumberInput = styled.input`
+  ${inputCSS}
+  padding: 0;
+  margin-right: 4px;
+  text-align: center;
+  width: 50%;
+  max-width: 50px;
+  ::-webkit-inner-spin-button,
+  ::-webkit-outer-spin-button {
+    margin: 0;
+    appearance: none;
+  }
+`;
+
+const TimeSeparator = styled.span`
+  padding: 0 3px 0 0;
+  color: #555d66;
+`;
+
+const Button = styled.button`
+  display: inline-flex;
+  text-decoration: none;
+  font-size: 13px;
+  margin: 0;
+  border: 0;
+  cursor: pointer;
+  transition: box-shadow 0.1s linear;
+  padding: 0 10px;
+  line-height: 2;
+  height: 30px;
+  border-radius: 3px;
+  white-space: nowrap;
+  border-width: 1px;
+  border-style: solid;
+  color: rgb(0, 117, 175);
+  border-color: rgb(0, 117, 175);
+  background: #f3f5f6;
+
+  &:focus {
+    outline: 2px solid transparent;
+    background: #f3f5f6;
+    color: rgb(0, 93, 140);
+    border-color: rgb(0, 118, 177);
+    box-shadow: 0 0 0 1px rgb(0, 118, 177);
+    text-decoration: none;
+  }
+  ${({ isToggled }) =>
+    isToggled &&
+    `
+  background: #edeff0;
+  border-color: #8f98a1;
+  box-shadow: inset 0 2px 5px -3px #555d66;
+  `}
+`;
+
+const InputGroup = styled.div`
+  flex-basis: 100%;
+`;
+
+const AMButton = styled(Button)`
+  margin-right: -1px;
+  border-radius: 3px 0 0 3px;
+`;
+
+const PMButton = styled(Button)`
+  margin-left: -1px;
+  border-radius: 0 3px 3px 0;
+`;
+
+function TimePicker({ onChange, is12Hour, localData, setLocalData }) {
```

Comment:
I'd like to see unit tests for this one - seems fairly straight-forward to do.

Reply:
The tests are currently purposefully left out since #1391 will most likely change all the time picking part. However, it should be quite straightforward indeed, will add!

Reply:
Added in 9495024

Reply:
I'd like to see an exhaustive unit test of this particular component too - it's fairly complex and has less then 50% coverage as it stands:

<img width="807" alt="Screen Shot 2020-05-21 at 12 00 27 PM" src="https://user-images.githubusercontent.com/637548/82578867-fd76ba80-9b5a-11ea-984c-4a2432fc8708.png">

Reply:
But we can also push this a bit down the line, if you think this component in particular will change a lot.

Reply:
I'm actually seeing this locally:
<img width="674" alt="Screenshot 2020-05-21 at 18 12 46" src="https://user-images.githubusercontent.com/3294597/82579905-befaf300-9b8e-11ea-9590-e75fcd894559.png">

This is mainly tested in the `assets/src/edit-story/components/inspector/document/` folder, in case you ran just one part. It's a bit wider than just a unit test.

Thoughts?

Reply:
Ah, I had forgotten to pull latest changes (because of local change it failed). But it still leaves branch coverage at 21/27 only. Some of the untested branches are trivial, but still.

Alright, let's proceed.

I just really like those 100% coverage numbers 😜.

Reply:
When the custom styling for the calendar comes, I promise I'll do my best to see it would have as close to 100% coverage as possible!